modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -4920,7 +4920,21 @@ private void _validateWorkflowAction(
 			  (status != WorkflowConstants.STATUS_DRAFT))) &&
 			(workflowAction == WorkflowConstants.ACTION_SAVE_DRAFT)) {
 
-			throw new ObjectEntryStatusException();
+			Locale locale = LocaleThreadLocal.getThemeDisplayLocale();
+
+			if (locale == null) {
+				locale = LocaleThreadLocal.getSiteDefaultLocale();
+			}
+
+			if (locale == null) {
+				User user = GuestOrUserUtil.getGuestOrUser();
+
+				locale = user.getLocale();
+			}
+
+			throw new ObjectEntryStatusException(_language.get(
```

Comment:
Maybe better to send directly the message in english. There is a story to improve the messaging in GraphQL. So get rid of the locale calculation and return better the english text "Draft status is not allowed" to simplify this.

This pattern is not being used in Liferay. There are other patterns like including the key (in this case, it would be "draft-status-is-not-allowed", but by know I think sending just the english text is enough")